aiida_mlip/data/model.py

```diff
@@ -0,0 +1,216 @@
+"""Define Model Data type in AiiDA."""
+
+import hashlib
+from pathlib import Path
+from typing import Any, Optional, Union
+from urllib import request
+from urllib.parse import urlparse
+
+from aiida.orm import SinglefileData
+
+
+class ModelData(SinglefileData):
+    """Class to save a model file as an AiiDA data type.
+
+    The file can be a file that is stored locally or a new file to download.
+    """
+
+    @staticmethod
+    def _calculate_hash(file: Union[str, Path]) -> str:
+        """Calculate the hash of a file.
+
+        Parameters
+        ----------
+        file : Union[str, Path]
+            Path to the file for which hash needs to be calculated.
+
+        Returns
+        -------
+        str
+            The SHA-256 hash of the file.
+        """
+        # Calculate hash
+        buf_size = 65536  # reading 64kB (arbitrary) at a time
+        sha256 = hashlib.sha256()
+        with open(file, "rb") as f:
+            # calculating sha in chunks rather than 1 large pass
+            while data := f.read(buf_size):
+                sha256.update(data)
+        file_hash = sha256.hexdigest()
+        return file_hash
+
+    @classmethod
+    def _check_existing_file(cls, file: Union[str, Path]) -> Path:
```

Comment:
now that I am looking one more time, what does it happen if you pass a relative path?